File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.kt
class Solution {
    fun solution(cards1: Array<String>, cards2: Array<String>, goal: Array<String>): String {
        val firstCard = cards1.toMutableList()
        val secondCard = cards2.toMutableList()

        for(i in goal.indices) {
            if(firstCard.size > 0 && firstCard[0] == goal[i]) {
                firstCard.removeAt(0)
            }else if(secondCard.size > 0 && secondCard[0] == goal[i]) {
                secondCard.removeAt(0)
            }else {
                return "No"
            }
        }
        return "Yes"
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.swift
func solution(_ cards1:[String], _ cards2:[String], _ goal:[String]) -> String {
    var card1Index = -1
    var card2Index = -1
    
    for i in 0...goal.count-1 {
        if cards1.contains(goal[i]) && card1Index+1 == cards1.firstIndex(of: goal[i]) ?? -1 {
            card1Index = cards1.firstIndex(of: goal[i])!
        } else if cards2.contains(goal[i]) && card2Index+1 == cards2.firstIndex(of: goal[i]) ?? -1 {
            card2Index = cards2.firstIndex(of: goal[i])!
        } else {
            return "No"
        }
    }
    
    return "Yes"
}